### PR TITLE
[CN-448] Update the tutorial for showing the enhanced use case of WanReplication

### PR DESCRIPTION
**Pull request by cagric0**

### CODE REVIEW

**Review by dzeromski-hazelcast — APPROVED** [review]

File: docs/modules/ROOT/pages/hazelcast-platform-operator-wan-replication.adoc [hunk 1]
```diff
@@ -14,7 +14,7 @@ In this tutorial, you'll do the following:
 
 - Deploy two Hazelcast clusters.
 
-- Create a Hazelcast map configuration.
+- Create two Hazelcast map configuration in one of the cluster.
```
Comment (angelasimms) [inline docs/modules/ROOT/pages/hazelcast-platform-operator-wan-replication.adoc]:
```suggestion
- Create two Hazelcast map configurations on one of the clusters.
```

File: docs/modules/ROOT/pages/hazelcast-platform-operator-wan-replication.adoc [hunk 2]
```diff
@@ -14,7 +14,7 @@ In this tutorial, you'll do the following:
...
 
 - Synchronize map data between two Hazelcast clusters.
```
Comment (angelasimms) [inline docs/modules/ROOT/pages/hazelcast-platform-operator-wan-replication.adoc]:
```suggestion
- Synchronize map data between the two Hazelcast clusters.
```
Also suggestions for the Before you Begin section below:

Before starting this tutorial, make sure that you have the following:
* A running https://kubernetes.io/[Kubernetes] cluster
* The Kubernetes command-line tool, https://kubernetes.io/docs/tasks/tools/#kubectl[`kubectl`]
* A deployment of Deployed xref:operator:ROOT:index.adoc[Hazelcast Platform Operator]

File: docs/modules/ROOT/pages/hazelcast-platform-operator-wan-replication.adoc [hunk 3]
```diff
@@ -75,7 +79,7 @@ EOF
 ----
 +
 
-. Check the status of the cluster to make sure that it's running.
+. Check the status of the clusters to make sure that it's running.
```
Comment (angelasimms) [inline docs/modules/ROOT/pages/hazelcast-platform-operator-wan-replication.adoc]:
```suggestion
. Check the status of the clusters to make sure that both clusters are running.
```

File: docs/modules/ROOT/pages/hazelcast-platform-operator-wan-replication.adoc [hunk 4]
```diff
@@ -108,23 +112,38 @@ The field `EXTERNAL-IP` is the address of your Hazelcast cluster.
 
 == Step 2. Create WAN Replication Configuration
 
-. Create a map in the first cluster.
+. Create two map in the first cluster.
```
Comment (angelasimms) [inline docs/modules/ROOT/pages/hazelcast-platform-operator-wan-replication.adoc]:
```suggestion
. Create two maps on the first cluster. In this example, the following maps are created:

- `example-map-1`
- `example-map-2`.
```

File: docs/modules/ROOT/pages/hazelcast-platform-operator-wan-replication.adoc [hunk 5]
```diff
@@ -108,23 +112,38 @@ The field `EXTERNAL-IP` is the address of your Hazelcast cluster.
 
 == Step 2. Create WAN Replication Configuration
```
Comment (angelasimms) [inline docs/modules/ROOT/pages/hazelcast-platform-operator-wan-replication.adoc]:
```suggestion
== Step 2. Create a WAN Replication Configuration
```

File: docs/modules/ROOT/pages/hazelcast-platform-operator-wan-replication.adoc [hunk 6]
```diff
@@ -108,23 +112,38 @@ The field `EXTERNAL-IP` is the address of your Hazelcast cluster.
...
+kind: Map
+metadata:
+  name: example-map-2
 spec:
   hazelcastResourceName: hazelcast-first
 EOF
 ----
 
-. Create a WAN Replication Configuration. Use the second cluster as the target cluster to receive the WAN Replication events.
+. Create a WAN Replication Configuration.
```
Comment (angelasimms) [inline docs/modules/ROOT/pages/hazelcast-platform-operator-wan-replication.adoc]:
```suggestion
. Create the configuration for WAN replication:
```

File: docs/modules/ROOT/pages/hazelcast-platform-operator-wan-replication.adoc [hunk 7]
```diff
@@ -108,23 +112,38 @@ The field `EXTERNAL-IP` is the address of your Hazelcast cluster.
...
+
++
+- Use the first cluster as the source cluster by giving the cluster name as a resource to WAN Replication configuration.
```
Comment (angelasimms) [inline docs/modules/ROOT/pages/hazelcast-platform-operator-wan-replication.adoc]:
```suggestion
- Use the first cluster as the source cluster by adding its name as a resource in the WAN Replication configuration.
```

File: docs/modules/ROOT/pages/hazelcast-platform-operator-wan-replication.adoc [hunk 8]
```diff
@@ -108,23 +112,38 @@ The field `EXTERNAL-IP` is the address of your Hazelcast cluster.
...
+Giving the cluster name as a resource starts the WAN Replication for both of the maps created earlier.
```
Comment (angelasimms) [inline docs/modules/ROOT/pages/hazelcast-platform-operator-wan-replication.adoc]:
```suggestion
Adding the cluster name as a resource starts WAN replication for both the maps that you created earlier.
```

File: docs/modules/ROOT/pages/hazelcast-platform-operator-wan-replication.adoc [hunk 9]
```diff
@@ -108,23 +112,38 @@ The field `EXTERNAL-IP` is the address of your Hazelcast cluster.
...
++
+- Use the second cluster as the target cluster to receive the WAN Replication events.
```
Comment (angelasimms) [inline docs/modules/ROOT/pages/hazelcast-platform-operator-wan-replication.adoc]:
```suggestion
- Add the second cluster as the target cluster to receive the WAN Replication events.
```

File: docs/modules/ROOT/pages/hazelcast-platform-operator-wan-replication.adoc [hunk 10]
```diff
@@ -108,23 +112,38 @@ The field `EXTERNAL-IP` is the address of your Hazelcast cluster.
...
+
 +
 Run the following command to configure the replication
```
Comment (angelasimms) [inline docs/modules/ROOT/pages/hazelcast-platform-operator-wan-replication.adoc]:
```suggestion
Run the following command to apply the configuration.
```

File: docs/modules/ROOT/pages/hazelcast-platform-operator-wan-replication.adoc [hunk 11]
```diff
@@ -215,8 +236,10 @@ client = hazelcast.HazelcastClient(
 --
 
 ====
+. Now start the application for each map with their name as a argument to fill the maps.
```
Comment (angelasimms) [inline docs/modules/ROOT/pages/hazelcast-platform-operator-wan-replication.adoc]:
```suggestion
. Now start the application for each map, using the map name as an argument to fill each map with random entries. If you're reusing the sample code from this tutorial, use the map names `example-map-1` and `example-map-2`.
```

File: docs/modules/ROOT/pages/hazelcast-platform-operator-wan-replication.adoc [hunk 12]
```diff
@@ -215,8 +236,10 @@ client = hazelcast.HazelcastClient(
...
 +
-Now you can start the application to fill the map.
+Map Names -> example-map-1, example-map-2
```
Comment (angelasimms) [inline docs/modules/ROOT/pages/hazelcast-platform-operator-wan-replication.adoc]:
```suggestion

```
Replaced by suggestion above.

File: docs/modules/ROOT/pages/hazelcast-platform-operator-wan-replication.adoc [hunk 13]
```diff
@@ -283,9 +306,9 @@ Current map size: 9
 Current map size: 10
 ----
 
-== Step 3. Verify that the Map Entries were Replicated
+== Step 3. Verify that the Maps' Entries were Replicated
```
Comment (angelasimms) [inline docs/modules/ROOT/pages/hazelcast-platform-operator-wan-replication.adoc]:
```suggestion
== Step 3. Verify the Replication of Map Entries
```

File: docs/modules/ROOT/pages/hazelcast-platform-operator-wan-replication.adoc [hunk 14]
```diff
@@ -283,9 +306,9 @@ Current map size: 9
...
 
-In this step, you verify the size of the map on the second cluster to make sure that it received all the WAN Replication events. 
+In this step, you verify the sizes of the maps on the second cluster to make sure that they received all the WAN Replication events.
```
Comment (angelasimms) [inline docs/modules/ROOT/pages/hazelcast-platform-operator-wan-replication.adoc]:
```suggestion
In this step, you'll check the sizes of the maps on the second, target cluster to make sure that WAN replication events have been received.
```

File: docs/modules/ROOT/pages/hazelcast-platform-operator-wan-replication.adoc [hunk 15]
```diff
@@ -283,9 +306,9 @@ Current map size: 9
...
 
 . Configure the Hazelcast client to connect to the second cluster as you did in <<configure-client, Configure the Hazelcast Client>>.
```
Comment (angelasimms) [inline docs/modules/ROOT/pages/hazelcast-platform-operator-wan-replication.adoc]:
```suggestion
. Configure the Hazelcast client to connect to the second cluster, as you did in <<configure-client, Configure the Hazelcast Client>>.
```

File: docs/modules/ROOT/pages/hazelcast-platform-operator-wan-replication.adoc [hunk 16]
```diff
@@ -361,7 +384,10 @@ client = hazelcast.HazelcastClient(
 ----
 --
 ====
-. Start the application to check the map size and see if the WAN Replication was successful.
+. Start the application for each map with their name as a argument to check the map size and see if the WAN Replication was successful.
```
Comment (angelasimms) [inline docs/modules/ROOT/pages/hazelcast-platform-operator-wan-replication.adoc]:
```suggestion
. Start the application for each map, using the map name as an argument to check the map size, and to check that WAN replication was successful. If you're reusing the sample code from this tutorial, use the map names `example-map-1` and `example-map-2`.
```

File: docs/modules/ROOT/pages/hazelcast-platform-operator-wan-replication.adoc [hunk 17]
```diff
@@ -361,7 +384,10 @@ client = hazelcast.HazelcastClient(
...
++
+Map Names -> example-map-1, example-map-2
```
Comment (angelasimms) [inline docs/modules/ROOT/pages/hazelcast-platform-operator-wan-replication.adoc]:
```suggestion

```
Included text in bullet point above.

File: docs/modules/ROOT/pages/hazelcast-platform-operator-wan-replication.adoc [hunk 18]
```diff
@@ -416,7 +442,7 @@ You should see the following output:
 [source, shell]
 ----
 Successful connection!
-Current map size: 12
+Current map (<MAP-NAME>) size: 12
 ----
 
 == Clean Up
```
Comment (angelasimms) [inline docs/modules/ROOT/pages/hazelcast-platform-operator-wan-replication.adoc:447]: Suggested updates to line 424:

To remove all custom resources and PVCs, run the following commands:

File: clients/go/main.go [hunk 19]
```diff
@@ -10,11 +10,11 @@ import (
 )
 
 func main() {
-	if len(os.Args) != 2 {
-		fmt.Println("You should pass an argument to run: fill or size")
+	if len(os.Args) != 3 {
+		fmt.Println("You should pass two argument. The First one is `fill` or `size`. The second one is map name.")
```
Comment (angelasimms) [inline clients/go/main.go]:
```suggestion
		fmt.Println("You need to pass two arguments. The first argument must be `fill` or `size`. The second argument must be `mapName`.")
```

File: clients/go/main.go [hunk 20]
```diff
@@ -10,11 +10,11 @@ import (
...
 		return
 	}
-	if !(os.Args[1] == "fill" || os.Args[1] == "size") {
+	if os.Args[1] != "fill" && os.Args[1] != "size" {
 		fmt.Println("Wrong argument, you should pass: fill or size")
```
Comment (angelasimms) [inline clients/go/main.go]:
```suggestion
		fmt.Println("Wrong argument, pass `fill` or `size` instead.")
```

File: clients/java/src/main/java/com/hazelcast/Main.java [hunk 21]
```diff
@@ -9,20 +9,22 @@
 
 public class Main {
     public static void main(String[] args) throws Exception {
-        if(args.length == 0) {
-            System.out.println("You should pass an argument to run: fill or size");
+        if(args.length != 2) {
+            System.out.println("You should pass two argument. The First one is `fill` or `size`. The second one is map name.");
```
Comment (angelasimms) [inline clients/java/src/main/java/com/hazelcast/Main.java]:
```suggestion
            System.out.println("You need to pass two arguments. The first argument must be `fill` or `size`. The second argument must be `mapName`.");
```

File: clients/nodejs/client.js [hunk 22]
```diff
@@ -12,21 +12,22 @@ const clientConfig = {
 
 (async () => {
     try {
-        if (process.argv.length === 2) {
-            console.error('You should pass an argument to run: fill or size');
+        if (process.argv.length !== 4) {
+            console.error('You should pass two argument. The First one is `fill` or `size`. The second one is map name.');
```
Comment (angelasimms) [inline clients/nodejs/client.js]:
```suggestion
            console.error('You need to pass two arguments. The first argument must be `fill` or `size`. The second argument must be `mapName`.');
```

File: clients/python/main.py [hunk 23]
```diff
@@ -6,8 +6,8 @@
 
 logging.basicConfig(level=logging.INFO)
 
-if len(sys.argv) != 2:
-    print("You should pass an argument to run: fill or size")
+if len(sys.argv) != 3:
+    print("You should pass two argument. The First one is `fill` or `size`. The second one is map name.")
```
Comment (angelasimms) [inline clients/python/main.py]:
```suggestion
    print("You need to pass two arguments. The first argument must be `fill` or `size`. The second argument must be `mapName`.")
```

**Review by angelasimms — CHANGES_REQUESTED** [review]
All changes are minor text updates.

File: docs/modules/ROOT/pages/hazelcast-platform-operator-wan-replication.adoc [hunk 24]
```diff
@@ -14,17 +14,17 @@ In this tutorial, you'll do the following:
...
+Before starting this tutorial, make sure that you have the following:
 
-* Up and running https://kubernetes.io/[Kubernetes] cluster
-* Kubernetes command-line tool, https://kubernetes.io/docs/tasks/tools/#kubectl[kubectl]
-* Deployed xref:operator:ROOT:index.adoc[Hazelcast Platform Operator]
+* A running https://kubernetes.io/[Kubernetes] cluster
+* The Kubernetes command-line tool, https://kubernetes.io/docs/tasks/tools/#kubectl[kubectl]
+* A deployed xref:operator:ROOT:index.adoc[Hazelcast Platform Operator]
 
 == Step 1. Start the Hazelcast Cluster
```
Comment (angelasimms) [inline docs/modules/ROOT/pages/hazelcast-platform-operator-wan-replication.adoc:30]: In this step, maybe you could just have the instruction below and remove the instruction on line 31.
```suggestion
. Create a secret with your link:http://trialrequest.hazelcast.com/[Hazelcast Enterprise License].
```

**Review by angelasimms — CHANGES_REQUESTED** [review]
I think there's some duplication at the start of Step 1.

**Review by angelasimms — APPROVED** [review]
All looks good to me. Thanks.

**Pull request merged**